src/main/java/com/redhat/rhjmc/containerjfr/net/web/handlers/TargetRecordingOptionsPatchHandler.java

```diff
@@ -0,0 +1,144 @@
+ * either these or other terms.
+ *
+ * This license is subject to the following condition:
+ * The above copyright notice and either this complete permission notice or at
+ * a minimum a reference to the UPL must be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ * #L%
+ */
+package com.redhat.rhjmc.containerjfr.net.web.handlers;
+
+import java.util.Arrays;
+import java.util.Map;
+
+import javax.inject.Inject;
+
+import org.openjdk.jmc.flightrecorder.configuration.recording.RecordingOptionsBuilder;
+
+import com.google.gson.Gson;
+
+import com.redhat.rhjmc.containerjfr.commands.internal.RecordingOptionsBuilderFactory;
+import com.redhat.rhjmc.containerjfr.core.RecordingOptionsCustomizer;
+import com.redhat.rhjmc.containerjfr.core.RecordingOptionsCustomizer.OptionKey;
+import com.redhat.rhjmc.containerjfr.net.AuthManager;
+import com.redhat.rhjmc.containerjfr.net.TargetConnectionManager;
+
+import io.vertx.core.MultiMap;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.ext.web.handler.impl.HttpStatusException;
+
+class TargetRecordingOptionsPatchHandler extends AbstractAuthenticatedRequestHandler {
+
+    static final String PATH = TargetRecordingOptionsGetHandler.PATH;
+    private final RecordingOptionsCustomizer customizer;
+    private final TargetConnectionManager connectionManager;
+    private final RecordingOptionsBuilderFactory recordingOptionsBuilderFactory;
+    private final Gson gson;
+
+    @Inject
+    TargetRecordingOptionsPatchHandler(
+            AuthManager auth,
+            RecordingOptionsCustomizer customizer,
+            TargetConnectionManager connectionManager,
+            RecordingOptionsBuilderFactory recordingOptionsBuilderFactory,
+            Gson gson) {
+        super(auth);
+        this.customizer = customizer;
+        this.connectionManager = connectionManager;
+        this.recordingOptionsBuilderFactory = recordingOptionsBuilderFactory;
+        this.gson = gson;
+    }
+
+    @Override
+    public HttpMethod httpMethod() {
+        return HttpMethod.PATCH;
+    }
+
+    @Override
+    public String path() {
+        return PATH;
+    }
+
+    @Override
+    void handleAuthenticated(RoutingContext ctx) throws Exception {
+        MultiMap attrs = ctx.request().formAttributes();
+        if (attrs.contains("toDisk")) {
+            try {
+                Boolean.parseBoolean(attrs.get("toDisk"));
```

Comment:
This doesn't quite work as intended - `parseBoolean` is `true` if the input string is `"true"`, case insensitive, and `false` for all other inputs. It doesn't throw.

On a related note, it would be good to exercise these validations with unit tests.

Reply:
I just realized that the POST handler should probably also do validation right?

Reply:
Yes, it probably should

Reply:
Ok, there are now tests for the validations I added. I also added validation for the boolean value in TargetRecordingPostHandler, I throw a NumberFormatException when the boolean can't be parsed to handle it together with the other exceptions, but I realize this is probably not the best way, Im just not sure what exception I should use.

Reply:
I agree that `NumberFormatException` seems like not quite the right fit here, although I don't think it's too important anyway - in any case, this will be an uncaught exception that is then later on caught by the `AbstractAuthenticatedRequestHandler` and turned into a response with status `500`, so all the exception type (and message) does is become part of the response status message. That will probably be displayed to end users in the end, but clients (as in, the web-client, operator, etc. - other automated clients) should only really care about the status code.

Maybe just `IllegalArgumentException` is fine, unless you want to create a custom one like `BooleanValueException` or `BooleanFormatException`. That would be nice, but it's probably overkill here.